### ISSUE TITLE
buffer slice

### ISSUE THREAD

**Issue by Riverzxz** [cloudwego/shmipc-go#49]
lock used for shmipc buffer slice? lock free??

**Comment by goodbye-babyer** [comment]
no. why you need write to buffer with concurrrently.  Maybe another solution is that  you coud create many streams and per stream's write operation could be concurrent
